### PR TITLE
#170947681 Allow travel admin to create accommodation facilities

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- Add functionality of creating accommodation facilities 
#### Description of Task to be completed?
- only travel admin will be able to create and edit accommodation facilities by providing `location name`,`streetNumber`,`number of rooms`,`type of rooms`and `building image`
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn   backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-create-accommodation-170947681` 
- run `npm install` to install dependencies
- check if you have PostgresQL database, if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run dropTables` to clean your database
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
- login by sending POST request to this url ``http://localhost:4000/api/v1/auth/login with the credentials I mentioned below
```
{
 "email":"traveladmin@gmail.com",
   "password": "Niyonkuru@1"
}
```
- create an accommodation facility by making POST request to this url `http://localhost:4000/api/v1/create/accommodation` along with this data 
```
{
 "accommodationName":"Hill top",
 "locationName": "Chicago",
 "streetNumber": "104st ",
   "numberOfRooms": 4,
   "description":"it has wifi and big secreen TV",
    "availableRooms": [
      {
        "roomName": "double room",
        "roomType": "king",
        "price":"$50/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$53/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$54/night"
      }
      ,
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$55/night"
      }
    ]
  }
```

- to view all the accommodations sent GET request to this URL `http://localhost:4000/api/v1/view/accommodations`
- to view specific accommodation make a GET request to this URL `http://localhost:4000/api/v1/view/accommodation/1` the number one `(1)` you see here is  accommodation `ID`, you can replace it with a another accommodation `ID`  you want to view
- to upload image of a specific accommodation use your postman as a form and name one field `imageOfBuilding` and give it file type in order to upload accommodation or center image, after uploading your image send PATCH request to this url`http://localhost:4000/api/v1/upload/accommodation/1`  `1` is ID of this accommodation we want to give an image
- if you want to edit a specific accommodation information make a PATCH request to this url `http://localhost:4000/api/v1/edit/accommodation/1`,  you can use the data provided below or write your own
```
{

    "availableRooms": [
      {
        "roomName": "double room",
        "roomType": "king",
        "price":"$50/night",
        "description":"it has wifi and big secreen TV"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$53/night",
        "description":"it has wifi and big secreen TV"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$54/night",
        "description":"it has wifi and big secreen TV"
      }
      ,
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$55/night",
        "description":"it has wifi and big secreen TV"
      }
    ]
  }
``` 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947681](https://www.pivotaltracker.com/story/show/170947681)
#### Create Accommodation
Screen Shot 2020-03-09 at 18 37 14

#### Upload image
Screen Shot 2020-03-07 at 17 50 02

#### View all accommodation
Screen Shot 2020-03-09 at 18 42 38

#### Edit accommodation
Screen Shot 2020-03-09 at 18 40 42

### CODE REVIEW

**Comment by Kagorora** [comment]
@EugeneMunya  don't you think it would be a great idea to include the name of the center/accomodation in your view all available accomodation. for instance the name of the hotel

File: src/controllers/accommodation.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import models from '../db/models';
+import dotenv from 'dotenv';
+dotenv.config();
```
Comment (kagabof) [inline src/controllers/accommodation.js]: please add two empty lines on after imports.

File: src/controllers/accommodation.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export default class accomodationFacility {
+  static async getAllAccommodations(req,res){
+    try{
+    const accommodations = await models.Accommodation.findAll();
+        return res.status(200).json({status:200, data:accommodations});
+    }catch(error){
```
Comment (kagabof) [inline src/controllers/accommodation.js]: Please fix your ESLint in order to have 2 spaces of indentation.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/controllers/accommodation.js (hunk 1 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js]: I have implemented it, for now, you can check

**Comment by EugeneMunya** [comment]
> @EugeneMunya don't you think it would be a great idea to include the name of the center/accomodation in your view all available accomodation. for instance the name of the hotel

I added that field and also the description field

**Review by kagabof — CHANGES_REQUESTED** [review]
Please check on those hound issues.

File: src/controllers/accommodation.js (hunk 1 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js]: I resolved the issue by configuring my eslint

File: src/controllers/accommodation.js (hunk 2 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js]: I have managed to configure my ESLint

**Review by kagabof — APPROVED** [review]
Great job.

**Pull request merged**